### PR TITLE
Fix Almost Everything

### LINKED ISSUE #18: Cleanup the markdown files
Some markdown files are still formatted poorly (a result of the migration from MediaWiki).

Any help in cleaning up the markdown documentation would be greatly appreciated.

This involves the following:
* Fix up formatting
* Fix broken links
* Add code tags where necessary
* If content is 404 - Not found, look in the internet-archive / waybackmachine and add the content back.

### PR DESCRIPTION
**Pull request by halotroop2288**
Almost every single compiler warning is fixed. (Eventually, we should be able to run it in strict mode!)

I did some manual linting (automatic linting would probably do more harm than good!)

- Hover text that just says "wikilink" has been removed.
- Some, if not all wiki links have had their alt text changed so they use plain English punctuation and spacing. (No more underscores or combining words together.)
- Image alt text that just says the file name has been removed, replaced with "Image"
  This could probably use the same treatment as other alt text, but it will take significantly longer to add alt text that is helpful.
- Some markdown tables have been aligned
- Some dead external links have been replaced (if I could find them)
- Added end-of-file newlines

This does touch every single file in the project, but there's no way around that if you want to fix #18.

Before merging:

- [x] Check that nothing has been deleted accidentally
      (Some things have been *moved*, but not deleted.)

### CODE REVIEW

**Comment by halotroop2288** [comment]
In the future:

- Redirects should be moved to MkDocs' [built-in redirect system](https://pypi.org/project/mkdocs-redirects/)
- Pages and images in categories should be moved to category folders
- Wiki navigation should be moved to MkDocs' built-in system
  This system supports either [nested categories](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#navigation-tabs) or [Table of Contents integration](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#navigation-integration)

**Comment by halotroop2288** [comment]
Ready for merge @tuxuser. I've reviewed my changes. LGTM!

**Comment by tuxuser** [comment]
You did your own review? :d kudos!
Yeah, due to lack of time I wasn't able to extensively review yours... But will do on this one. Hopefully today.

Thx alot for your contributions btw

File: docs/NAND.md [hunk 1]
```diff
@@ -104,17 +104,12 @@ This also works in reverse:
...
-As well, you can build your own USB SPI Flasher with [this
-diagram](http://img35.imageshack.us/img35/8949/xbox360usbflasher.png)
-and use [these
-binaries](http://nds.cmamod.com/x360/PICFLASH_v3b_plus2.zip). Flash the
-PicXboot.hex to your PIC, then use the programmer from [this
-picflash](http://www.megaupload.com/?d=X6TBNFDC) to flash the
+You can also wire LPT directly to the flash using this [diagram](http://img19.imageshack.us/img19/5198/wiringforxenondiagram.jpg).
+As well, you can build your own USB SPI Flasher with [this diagram](http://img35.imageshack.us/img35/8949/xbox360usbflasher.png)
+and use [these binaries](http://nds.cmamod.com/x360/PICFLASH_v3b_plus2.zip). Flash the
+PicXboot.hex to your PIC, then use the programmer from [this picflash](http://www.megaupload.com/?d=X6TBNFDC) to flash the
```
[the remark this answers is not in the document]
Comment (tuxuser) [reply docs/NAND.md:110]: We got em here: https://sourceforge.net/projects/free60/files/PicFlash/

[the remark this answers is not in the document]
Comment (halotroop2288) [reply docs/NAND.md:110]: That kinda leaves me more confused about what it means by "the programmer from this picflash" but maybe it means an exe in the zip?

Anyway, I've updated the "these binaries" link and removed the "this picflash" link as they appear to be the same thing.

**Comment by halotroop2288** [comment]
> Thx alot for your contributions btw

No problem! I love contributing to documentation, and this wiki needs a lot of help to be great again.
I often use it for my (@xenia-project related) projects.

**Pull request merged**